### PR TITLE
[luasocket] bump to 3.1.0

### PR DESCRIPTION
**Pull request by TheCycoONE**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/luasocket/portfile.cmake [hunk 1]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO lunarmodules/luasocket
-    REF v3.0.0
-    SHA512 4f93d6c0b602333df50ee4f939cd0419243f6de333472ffebf99334e301143e8cdee3bc1655c29f81608622d6e7850a9bcf6929a6d4748210a70cdb8218a1ec6
+    REF v3.1.0
```
Comment (JonLiu1993) [inline ports/luasocket/portfile.cmake:4]:
```suggestion
    REF "v{VERSION}"
```

File: ports/luasocket/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "luasocket",
-  "version": "3.0.0",
+  "version": "3.1.0",
   "port-version": 1,
```
Comment (JonLiu1993) [inline ports/luasocket/vcpkg.json:4]:
```suggestion
```

**Comment by TheCycoONE** [comment]
Updated with the suggestions from @JonLiu1993 
Thanks.

**Comment by TheCycoONE** [comment]
@JonLiu1993 @Cheney-W is anything else expected of me?

File: ports/luasocket/portfile.cmake [hunk 3]
```diff
@@ -18,28 +18,19 @@ vcpkg_copy_pdbs()
...
-    if(EXISTS "${CURRENT_PACKAGES_DIR}/bin/socket/socket.core.pdb")
-        file(RENAME "${CURRENT_PACKAGES_DIR}/bin/socket/socket.core.pdb" "${CURRENT_PACKAGES_DIR}/bin/socket/core.pdb")
-    endif()
-    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/bin/socket/socket.core.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/socket/core.dll")
-    if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/bin/socket/socket.core.pdb")
-        file(RENAME "${CURRENT_PACKAGES_DIR}/debug/bin/socket/socket.core.pdb" "${CURRENT_PACKAGES_DIR}/debug/bin/socket/core.pdb")
-    endif()
+    file(RENAME "${CURRENT_PACKAGES_DIR}/bin/socket/socket.core.dll" "${CURRENT_PACKAGES_DIR}/bin/socket/core.dll" RESULT temp)
+    file(RENAME "${CURRENT_PACKAGES_DIR}/bin/socket/socket.core.pdb" "${CURRENT_PACKAGES_DIR}/bin/socket/core.pdb" RESULT temp)
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/bin/socket/socket.core.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/socket/core.dll" RESULT temp)
```
Comment (JavierMatosD) [inline ports/luasocket/portfile.cmake:26]: I see you are storing the results of the rename in `temp`, but  I don't see `temp` being used for anything. Is this intentional or am I missing something?

**Comment by TheCycoONE** [comment]
It's intentional to avoid failing when the files don't exist. There use to
be conditionals on them and even they didn't cover community triplets.

**Comment by TheCycoONE** [comment]
https://cmake.org/cmake/help/latest/command/file.html#rename

**Comment by dg0yt** [comment]
> It's intentional to avoid failing when the files don't exist. There use to be conditionals on them and even they didn't cover community triplets.

A variable name like `unused` would have made this clear. 
And I think at least for the DLLs it would have been good to have proper conditionals.

**Pull request merged**